contracts/near/eth-client/src/lib.rs

```diff
@@ -105,27 +109,44 @@ pub struct EthClient {
     trusted_signer: Option<AccountId>,
     /// Mask determining all paused functions
     paused: Mask,
+    /// Store the bsc epoch header key 
+    epoch_header: H256,
+    /// chain id
+    chain_id: u64,
```

Comment:
Since the state of the contract is changing we can't redeploy on top of the current contracts as is.

We need to migrate the state first! Some resources:
- https://stackoverflow.com/a/68191607/4950797
- https://github.com/near/NEPs/pull/123

We either need to do this atomically or pause the contract before hand, to avoid inconsistencies.

Reply:
This is needed only because we don't want to maintain two different versions of the client, though we are having two different versions deployed one for Ethereum and one for BSC.